### PR TITLE
azurerm_point_to_site_vpn_gateway - support `route` block

### PR DESCRIPTION
**Pull request by njuCZ**
image

### CODE REVIEW

File: azurerm/internal/services/network/point_to_site_vpn_gateway_resource.go [hunk 1]
```diff
@@ -93,6 +93,48 @@ func resourceArmPointToSiteVPNGateway() *schema.Resource {
 								},
 							},
 						},
+
+						"route_config": {
```
Comment (katbyte) [inline azurerm/internal/services/network/point_to_site_vpn_gateway_resource.go:97]: can we drop config from here? 
```suggestion
						"route": {
```

File: azurerm/internal/services/network/point_to_site_vpn_gateway_resource.go [hunk 2]
```diff
@@ -93,6 +93,48 @@ func resourceArmPointToSiteVPNGateway() *schema.Resource {
...
+														ValidateFunc: validate2.VirtualHubRouteTableID,
+													},
+												},
+
+												"labels": {
+													Type:     schema.TypeList,
+													Optional: true,
+													Elem: &schema.Schema{
+														Type: schema.TypeString,
+													},
```
Comment (katbyte) [inline azurerm/internal/services/network/point_to_site_vpn_gateway_resource.go:130]: could we validate this?

File: azurerm/internal/services/network/point_to_site_vpn_gateway_resource.go [hunk 3]
```diff
@@ -93,6 +93,48 @@ func resourceArmPointToSiteVPNGateway() *schema.Resource {
...
+													Type:     schema.TypeList,
+													Required: true,
+													Elem: &schema.Schema{
+														Type:         schema.TypeString,
+														ValidateFunc: validate2.VirtualHubRouteTableID,
+													},
+												},
+
+												"labels": {
+													Type:     schema.TypeList,
```
Comment (katbyte) [inline azurerm/internal/services/network/point_to_site_vpn_gateway_resource.go:126]: shouldn't this be a set?

File: azurerm/internal/services/network/point_to_site_vpn_gateway_resource.go [hunk 4]
```diff
@@ -103,6 +145,46 @@ func resourceArmPointToSiteVPNGateway() *schema.Resource {
 				ValidateFunc: validation.IntAtLeast(0),
 			},
 
+			"custom_dns_servers": {
```
Comment (katbyte) [inline azurerm/internal/services/network/point_to_site_vpn_gateway_resource.go:148]: i'm not sure if custom is really necessary here as its implied?

File: website/docs/r/point_to_site_vpn_gateway.html.markdown [hunk 5]
```diff
@@ -52,18 +54,50 @@ A `connection_configuration` block supports the following:
...
+
+* `propagated_route_table` - (Optional) A `propagated_route_table` block as defined below.
+
+---
+
+A `propagated_route_table` block supports the following:
+
+* `ids` - (Required) The list of Virtual Hub Route Table resource id which the routes will be propagated to.
+
+* `labels` - (Optional) The list of labels to logically group VWan Route Tables.
```
Comment (katbyte) [inline website/docs/r/point_to_site_vpn_gateway.html.markdown:79]: Could we fix this up?

File: azurerm/internal/services/network/point_to_site_vpn_gateway_resource.go [hunk 6]
```diff
@@ -103,6 +145,46 @@ func resourceArmPointToSiteVPNGateway() *schema.Resource {
...
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.IsIPv4Address,
+				},
+			},
+
+			"connection_health": {
```
Comment (katbyte) [inline azurerm/internal/services/network/point_to_site_vpn_gateway_resource.go:157]: i don't think any of this makes sense in terraform so could we remove them? they will constantly be changed

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @njuCZ - i've left some comments inline that need to be addressed

File: azurerm/internal/services/network/point_to_site_vpn_gateway_resource.go (hunk 2 above)
Comment (njuCZ) [reply azurerm/internal/services/network/point_to_site_vpn_gateway_resource.go:131]: I am sorry I could not try out the validation rule for this field, just add a basic validation: `ValidateFunc: validation.StringIsNotEmpty`

**Comment by njuCZ** [comment]
@katbyte thanks for your review, I have updated this PR, could you have a look again?

**Review by katbyte — APPROVED** [review]
Thanks @njuCZ - LGTM 👍

**Pull request merged**